swim/suspicion.go

```diff
@@ -33,62 +33,85 @@ type suspect interface {
 	incarnation() int64
 }
 
+type timer struct {
+	*time.Timer
+	// at which state the timer was started
+	state string
+}
+
 // Suspicion handles the suspicion sub-protocol of the SWIM protocol
 type suspicion struct {
```

Comment:
I didn't suggest changing the name of the struct. Only to keep it called a "timer" further down in the patch.